disableBuildPeriodicalyTriggerFreestyleProjects.groovy

```diff
@@ -0,0 +1,29 @@
+/**
+The script was created on 2019-07-21
+by Kirill Gostaf, kgostaf@cloudbees.com
+
+This script will iterate over all the projects, and check whether a job has Build periodically trigger enabled.
+If the project type is Freestyle job the Build periodically trigger will be disabled.
+Otherwise, it prints the name of the project without modifying the trigger settings.
+For a pipeline project, i.e. org.jenkinsci.plugins.workflow.job.WorkflowJob, you may want to modify
+the Jenkins file that contains `triggers { cron(H/4 * * * *) }` instructions.
+*/
+
+
+import hudson.model.*
+import hudson.triggers.*
+TriggerDescriptor TIMER_TRIGGER_DESCRIPTOR = Hudson.instance.getDescriptorOrDie(TimerTrigger.class)
+
+for(item in Jenkins.instance.getAllItems(Job))
+{
+  def timertrigger = item.getTriggers().get(TIMER_TRIGGER_DESCRIPTOR)
+  if (timertrigger) {
+    if (item.class.canonicalName == "hudson.model.FreeStyleProject") {
```

Comment:
I think you could have also used:

```
if(item instanceof FreeStyleProject) {
 ...
}
```

Or, you can filter the items by doing:

```
Jenkins.instance.getAllItems(FreeStyleProject)
```

Or

```
Jenkins.instance.getAllItems(Job)
.findAll {it instanceof FreeStyleProject}
```

Reply:
Yes, I like (item instanceof FreeStyleProject).
The goal was to iterate over all type of projects, and to identify
all the projects where **Build periodically trigger** is enabled.
For pipeline project, the right place to disable Build by trigger would
be via Jenkinsfile. This script just points out if there exists any
pipeline projects.